queue/queue.go

```diff
@@ -22,6 +25,18 @@ const (
 
 type Store struct {
 	*bolt.DB
+	pubsub.Publisher
+}
+
+func (db *Store) publishResponseJSON(resp mdm.Response) error {
+	buf := new(bytes.Buffer)
+	enc := json.NewEncoder(buf)
```

Comment:
Why are you converting something that came as a plist to json? The conversion is lossy between the two 

IMO you should just take the []byte coming from the http request to /connect and pubslish it there.